src/metaxy/models/filter_expression.py

```diff
@@ -0,0 +1,409 @@
+"""SQL filter string parsing into Narwhals expressions.
+
+This module exposes Pydantic models that represent backend-agnostic filter
+expressions and utilities to parse SQL WHERE-like strings into Narwhals
+``IntoExpr`` objects. The primary entry point is :func:`parse_filter_string`,
+which returns a Narwhals expression that can be fed directly into
+``LazyFrame.filter`` (works across all Narwhals backends).
+"""
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import Annotated, Any, ClassVar, Literal
+
+import narwhals as nw
+from pydantic import Field, field_validator
+from sqlglot import exp, parse_one
```

Comment:
maybe keep it as `import sqlglot`, this is the most important lib in this file and it should be clear where it is used vs our functions